src/connection.js

```diff
@@ -715,7 +715,9 @@ class Connection extends EventEmitter {
   socketClose() {
     this.debug.log('connection to ' + this.config.server + ':' + this.config.options.port + ' closed');
     if (this.state === this.STATE.REROUTING) {
-      this.debug.log('Rerouting to ' + this.routingData.server + ':' + this.routingData.port);
+      if (typeof this.routingData !== 'undefined') {
```

Comment:
This change here seems unrelated? 🤔 If it's not related to the other changes, could you revert this?

Reply:
Ah, no, I see that you're re-using the re-routing logic here. I'm not sure how I feel about this, but if that's working, why not? 👍

Reply:
Yeah I had the same feeling if I'm honest, but I traded that against the scale of change I'd need to do to encapsulate a 'reconnect' state change :/